### ISSUE TITLE
Mismatch between Readme and Code regard Total Supply

### ISSUE THREAD

**Issue by PwC-SCA** [WETH10/WETH10#96]
The [Readme](https://github.com/WETH10/WETH10/blob/269b367e49bbf58bfb722ea1419743180ef0cea4/README.md#total-supply) says:

```md
Total Supply
The supply of WETH10 is capped at type(uint112).max.
```

However, the code defines it as:
https://github.com/WETH10/WETH10/blob/269b367e49bbf58bfb722ea1419743180ef0cea4/contracts/WETH10.sol#L72-L74

As `flashMinted` alone can reach `type(uint112).max`:
https://github.com/WETH10/WETH10/blob/269b367e49bbf58bfb722ea1419743180ef0cea4/contracts/WETH10.sol#L139

It seems that `address(this).balance + flashMinted` can exceed `type(uint112).max`.

**Comment by albertocuestacanada** [comment]
True, the cap was modified to simplify the code. Total supply is the ETH balance of the contract, plus any amount flash minted if we are in the middle of a flash mint.

The flash mints themselves are capped at `type(uint112).max`.

Would you like to open a PR to fix this?
